packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { DataObjectWithCounter, dataObjectWithCounterFactory } from "./dataObjectWithCounter";
+import { allFactories } from "./testDataObjects";
+
+const childKey = "childKey";
+export class DataObjectWithChildDataObject extends DataObjectWithCounter {
+    private unrefTimeStampMs?: number;
+    private oldChild?: DataObjectWithCounter;
+    private child?: DataObjectWithCounter;
+
+    public static get type(): string {
+        return "DataObjectWithChildDataObject";
+    }
+
+    protected async initializingFirstTime(props?: any): Promise<void> {
+        await super.initializingFirstTime(props);
+        await this.referenceNewChild();
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        await super.hasInitialized();
+        const childHandle = this.root.get<IFluidHandle<DataObjectWithCounter>>(childKey);
+        this.child = await childHandle?.get();
+    }
+
+    public stop() {
```

Comment:
I am not sure I understand how these functions will be used. Will the creator of this data store control when to create / reference / unreference a child? The reference / unreference functions are public which is leading me to believe that's the case.
I would imagine once this data store is created and initialized, it will start doing its thing where it does a certain action after a certain period of time. All the creator does is starts / stops this guy.
Infact, it seems like this is exactly what this data store does for its child. It's only missing a `start` call to its child when the child is referenced. Without that, the child would only do activity until the first `stop` call.

Reply:
Method removed, may revisit at a later time